### PR TITLE
Improve module objectfiles handling and moduleonly package

### PR DESCRIPTION
**Pull request by Arthapz**
this patch improve modules support by
    - ensuring that static / dynamic libraries public module objectfiles are beeing packed in the library
    - improving module culling so no std or std.compat module should be built if they are not imported
    - showing a warning when modules that are not std or std.compat are culled
    - raising an error when duplicated module name are detecteds

### CODE REVIEW

**Comment by Arthapz** [comment]
need https://github.com/xmake-io/xmake/pull/5375

File: xmake/rules/c++/modules/modules_support/dependency_scanner.lua [hunk 1]
```diff
@@ -200,7 +200,9 @@ function _get_package_modules(target, package, opt)
     for _, metafile in ipairs(metafiles) do
         package_modules = package_modules or {}
         local modulefile, name, metadata = _parse_meta_info(target, metafile)
-        package_modules[name] = {file = path.join(modulesdir, modulefile), metadata = metadata}
+        local package_data = package["_INFO"]
```
Comment (waruqi) [inline xmake/rules/c++/modules/modules_support/dependency_scanner.lua:203]: please do not access private vars. we should call `package:libraryfiles()` or `package:get()` to get them.

**Comment by Arthapz** [comment]
https://github.com/xmake-io/xmake/pull/5240 will no longer be necessary as this test is included in this PR

**Comment by waruqi** [comment]
> #5240 will no longer be necessary as this test is included in this PR

I will close it when this patch is merged.

File: tests/projects/c++/modules/duplicate_name_detection/test.lua [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+function _build()
+    try {function() 
+        if ci_is_running() then
+            os.iorun("xmake -rvD")
+        else
+            os.iorun("xmake -r")
+        end
+    end, catch {function(errors)
+        errors = errors.errors
```
Comment (waruqi) [inline tests/projects/c++/modules/duplicate_name_detection/test.lua:13]:
```lua
errors = tostring(errors)
```

**Pull request merged**